packages/brain/src/modules/cron/trackValidatorsPerformance/insertPerformanceData.ts

```diff
@@ -33,42 +38,91 @@ export async function insertPerformanceDataNotThrow({
   validatorsAttestationsTotalRewards: TotalRewards[];
   executionClient: ExecutionClient;
   consensusClient: ConsensusClient;
-  error?: Error;
+  error?: ValidatorPerformanceError;
 }): Promise<void> {
   for (const validatorIndex of activeValidatorsIndexes) {
-    //const liveness = validatorsLiveness.find((liveness) => liveness.index === validatorIndex)?.is_live;
+    if (error) {
+      await insertDataNotThrow({
+        postgresClient,
+        validatorPerformance: {
+          validatorIndex: parseInt(validatorIndex),
+          epoch: currentEpoch,
+          executionClient,
+          consensusClient,
+          error
+        }
+      });
+      continue;
+    }
+
     const attestationsTotalRewards = validatorsAttestationsTotalRewards.find(
       (attestationReward) => attestationReward.validator_index === validatorIndex
     );
-
     if (!attestationsTotalRewards) {
-      logger.error(`${logPrefix}Missing data for validator ${validatorIndex}, att: ${attestationsTotalRewards}`);
+      await insertDataNotThrow({
```

Comment:
why insert data and skip the rest of this validator's loop before `blockProposalStatus` is checked? `blockProposalStatus` could be okay and we wouldnt write it on db.

I would first try to fetch all data of a validator (proposal status & att rewards) and only then decide:
- If one of both is null, dont write anything, only error
- If one of both is null, write the one that isnt null and also an error